system_setup/common/wsl_conf.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python3
+# Copyright 2015-2021 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#    original code from ubuntuwslctl.core.loader
+#    Copyright (C) 2021 Canonical Ltd.
+
+import os
+import logging
+from configparser import ConfigParser
+
+log = logging.getLogger("system_setup.common.wsl_conf")
+
+config_base_default = {
+    "wsl": {
+        "automount": {
+            "root": "/mnt/",
+            "options": ""
+        },
+        "network": {
+            "generatehosts": "true",
+            "generateresolvconf": "true"
+        }
+    }
+}
+
+config_adv_default = {
+    "wsl": {
+        "automount": {
+            "enabled": "true",
+            "mountfstab": "true"
+        },
+        "interop": {
+            "enabled": "true",
+            "appendwindowspath": "true"
+        }
+    },
+    "ubuntu": {
+        "GUI": {
+            "theme": "default",
+            "followwintheme": "false"
+        },
+        "Interop": {
+            "guiintegration": "false",
+            "audiointegration": "false",
+            "advancedipdetection": "false"
+        },
+        "Motd": {
+            "wslnewsenabled": "true"
+        }
+    }
+}
+
+
+def wsl_config_loader(data, pathname, config_ref, id):
+    """
+    Loads the configuration from the given file type,
+    section and reference config.
+
+    :param data: dict, the data to load into
+    :param pathname: string, the path to the file to load
+    :param id: string, the name of the section to load
+    """
+    if not os.path.exists(pathname):
+        return data
+    config = ConfigParser()
+    config.read(pathname)
+    for conf_sec in config:
+        if conf_sec in config_ref[id]:
+            conf_sec_list = config[conf_sec]
+            for conf_item in conf_sec_list:
+                if conf_item in config_ref[id][conf_sec]:
+                    data[conf_sec.lower()
+                         + "_" + conf_item.lower()] = \
+                             conf_sec_list[conf_item]
+    return data
+
+
+def default_loader(is_advanced):
```

Comment:
Suggestion: make it a key paramter with default value. That way, the calls would become:
`default_loader(is_advanced=True)` which is more explicit than `default_loader(True)`

Reply:
done